tests/unit/test_cmr_search.py

```diff
@@ -399,3 +403,108 @@ def test_requests_error(self, mock_requests_get):
+        mock_response = Mock(spec=requests.Response)
+        mock_response.json.return_value = {'access_token': 'edl-token'}
+        mock_requests_post.return_value = mock_response
+
+        # Input parameters
+        urs_uat_edl_token_endpoint = UrsEnvsEdlTokenEndpoints.get(CMR_UAT)
+        edl_token_from_launchpad_response = get_edl_token_from_launchpad(
+            self.launchpad_token_header, CMR_UAT)
+
+        self.assertEqual(edl_token_from_launchpad_response, 'edl-token')
+
+        # Check if `get_edl_token_from_launchpad` was called once
+        # with expected parameters.
+        mock_requests_post.assert_called_once_with(
+            url=urs_uat_edl_token_endpoint,
+            data=f'token={self.launchpad_token_header}',
+            timeout=10)
+
+
+    @patch('requests.post')
+    def test_success_edl_token_from_launchpad_response(self,
+                                                       mock_requests_post):
+        ''' Check if the `get_edl_token_from_launchpad` response contains
+            the expected content for a successful response.
+        '''
+        # Create `mock_requests_post` for a successful request
+        # and set its return_value
+        mock_response = Mock(spec=requests.Response)
+        mock_response.ok = True
+        mock_response.json.return_value = {'access_token': 'edl-token'}
+        mock_requests_post.return_value = mock_response
+
+        # Test successful request
+        successful_response = get_edl_token_from_launchpad(
+            self.launchpad_token_header, CMR_UAT)
+
+        self.assertEqual(successful_response, 'edl-token')
+
+    @patch('requests.post')
+    def test_bad_edl_token_from_launchpad_response(self, mock_requests_post):
+        ''' Check if the `get_edl_token_from_launchpad_response` contains
+            the expected content for unsuccessful response.
+        '''
+        # Create `mock_requests_post` for a unsuccessful request
+        # and set its return_value
+        mock_response = Mock(spec=requests.Response)
+        mock_response.status_code = 400
+        mock_response.raise_for_status.side_effect = HTTPError(response=mock_response)
+        mock_requests_post.return_value = mock_response
+
+        with self.assertRaises(GetEdlTokenException):
+            get_edl_token_from_launchpad(self.launchpad_token_header, CMR_UAT)
+
+
+    @patch('requests.post', side_effect=Exception('Request timed out'))
+    def test_request_exception(self, mock_requests_post):
+        ''' Check if `GetEdlTokenException` is raised if `requests.post`
+            fails.
+        '''
+        # Create `mock_requests_post` for a unsuccessful request
+        # and set its return_value and side effect
+        mock_response = Mock(spec=requests.Response)
+        mock_response.status_code = 400
+        mock_response.raise_for_status.side_effect = Timeout(response=mock_response)
+        mock_requests_post.return_value = mock_response
```

Comment:
Can you elaborate on this? Do you mean you don't think I need this test or that I should update it to your suggestion?

Reply:
So on L469, you're patching the requests.post to have a side effect that raises an exception.  And then here, you also setting the return value to raise an exception, but that's not getting used.  

If you comment out the side_effect in the @patch on L469, this test doesn't pass, because the Timeout on L478 doesn't contain the "Request timed out" string.


``` shell
>       self.assertEqual(str(context_manager.exception),
                         str(GetEdlTokenException('Request timed out')))
E       AssertionError: 'requests module failed with the following error: ' != 'requests module failed with the following error: Request timed out'
E       - requests module failed with the following error: 
E       + requests module failed with the following error: Request timed out
E       ?                                                  +++++++++++++++++

```

So I would say you should just have requests.post have a side effect and remove all of the stuff I mentioned as being extraneous.

Reply:
Ahh makes sense! Thanks for the clarification:
* [ ] Update to above